reco_utils/dataset/download_utils.py

```diff
@@ -79,3 +82,43 @@ def download_path(path=None):
     else:
         path = os.path.realpath(path)
         yield path
+
+def get_blob_service(azure_storage_account_name, azure_storage_sas_token, container_name):
+    """ Get the Azure blob service for accessing the dataset.
+    
+    Args:
+        azure_storage_account_name (str): Azure storage account name.
+        azure_storage_sas_token (str): Azure storage SaS token.
+        container_name (str): Azure storage container name.
+    
+    Returns:
+        blob_service (azure.storage.blob.BlockBlobService): Azure BlockBlobService for dataset.
+    """
+
+    # create a blob service
+    blob_service = BlockBlobService(
+        account_name=azure_storage_account_name,
+        sas_token=azure_storage_sas_token,
+    )
+    
+    return blob_service
+
+def load_csv_from_blob(blob_service, container_name, blob_path):
```

Comment:
I see. Do you suggest instead to move this back into `covid_utils.py`?

Also, @miguelgfierro how do you feel about keeping the Azure blob storage interaction code just in `covid_utils.py` and no longer in `download_utils.py`?

Reply:
>  I think it will be better if users manage how the data is retrieved and loaded. For example it may be undesirable to load the entire blob into memory and duplicate into a DataFrame. Also it is quite common for arguments to be passed into the read_csv function, or for specific parsers to be used.

not sure I understand this, can you elaborate?

Reply:
my thought was it's only two lines of code, but it locks the user into one way of loading a dataframe from blob csv. I'm curious if it's worth wrapping in a separate function?

if we want to keep this and other blob utils? I like the idea of having a blob_utils module and this can be generalized and potentially more memory efficient using a stream. plus we can pass any specific kwargs for read_csv down like this.

```
def load_csv_from_blob(blob_service, container_name, blob_path, df_kwargs):
    """ Load a Pandas DataFrame from CSV in Azure Blob storage"""

    with StringIO() as stream:
        blob_service.get_blob_to_stream(container_name=container_name, blob_path=blob_path, stream=stream)
        df = pd.read_csv(stream, **df_kwargs)
    return df
```

Reply:
this is really nice, I would suggest this change:
```
def load_csv_from_blob(blob_service, container_name, blob_name,  **kwargs):
...
```
this should allow adding an arbitrary number of arguments

Reply:
Great suggestions! I have moved `load_csv_from_blob()` into a new file called `blob_utils.py` and have added **kwargs.

Unfortunately, using a stream with `StringIO()` doesn't seem to work with `get_blob_from_stream()` https://github.com/Azure/azure-storage-python/issues/536

I played around with the code trying to get it to work but wasn't able to, so I decided to keep what we had (`df = pd.read_csv(StringIO(blob.content), **kwargs)`).

Reply:
ah that's too bad streaming isn't supported yet, ok, sounds fine